### PR TITLE
chore: Add support for Scala 2.13.15

### PR DESCRIPTION
**Pull request by tgodzik**
Fixes https://github.com/scalameta/scalameta/issues/3973

### CODE REVIEW

File: tests/jvm/src/test/scala/scala/meta/tests/semanticdb/OccurrenceSuite.scala [hunk 1]
```diff
@@ -22,12 +22,8 @@ class OccurrenceSuite extends FunSuite {
 
   private def testBody(body: OccurrenceSuite.TestBody): Unit = {
     val expectedCompat =
-      if (ScalaVersion.atLeast212_14) body.expected
-      else
-        // Predef.type etc. was fixed in 2.12.14
-        body.expected.replace("Predef/*=>scala.Predef.*/.type", "Predef.type")
```
Comment (kitbellew) [inline tests/jvm/src/test/scala/scala/meta/tests/semanticdb/OccurrenceSuite.scala:28]: is this unrelated? cleanup for 2.12?

File: tests/jvm/src/test/resources/metac.expect_2.13 [hunk 2]
```diff
@@ -1729,11 +1729,6 @@ Synthetics:
       a,
       b
     ) == (1, 2))}).flatMap[Tuple6[Int, Int, Int, Int, Tuple4[Int, Int, Int, Int], Tuple4[Int, Int, Int, Int]]]({(b) => orig(List((a, b))).withFilter({(check$ifrefutable$1) => orig()}).withFilter({(local7) => orig((
-      a,
```
Comment (kitbellew) [inline tests/jvm/src/test/resources/metac.expect_2.13:1732]: why did this disappear?

**Review by kitbellew — APPROVED** [review]
i am assuming you understand the changes, so asked anyway...

File: tests/jvm/src/test/scala/scala/meta/tests/semanticdb/OccurrenceSuite.scala (hunk 1 above)
Comment (tgodzik) [reply tests/jvm/src/test/scala/scala/meta/tests/semanticdb/OccurrenceSuite.scala:28]: Yep, that's not necessary anymore and I was already cleaning this up.

File: tests/jvm/src/test/resources/metac.expect_2.13 (hunk 2 above)
Comment (tgodzik) [reply tests/jvm/src/test/resources/metac.expect_2.13:1732]: I would assume that some internal desugarings changed, but I honestly think these synthetics are not being used currently by anyone.

**Pull request merged**